evalml/tests/automl_tests/test_automl.py

```diff
@@ -5480,3 +5490,191 @@ def test_holdout_set_results_and_rankings(caplog, AutoMLTestEnv):
+        ValueError,
+        match="Objectives to include from the recommendation score should be a list",
+    ):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation="Not a list",
+        )
+    with pytest.raises(
+        ValueError,
+        match="Objectives to exclude from the recommendation score should be a list",
+    ):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            exclude_recommendation="Not a list",
+        )
+    with pytest.raises(ObjectiveNotFoundError, match="not a valid Objective!"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation=["Does not exist"],
+        )
+    with pytest.raises(ValueError, match="not defined for binary"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            include_recommendation=["R2"],
+        )
+    with pytest.raises(ValueError, match="Cannot exclude objective"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            use_recommendation=True,
+            exclude_recommendation=["Precision"],
+        )
+    AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        use_recommendation=True,
+        include_recommendation=["F1"],
+    )
+    assert "already one of the default objectives" in caplog.text
+
+
+def test_recommendation_include_non_optimization(X_y_binary):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        use_recommendation=True,
+        include_recommendation=["Recall"],
+    )
+    assert "Recall" in automl.recommendation_objectives
+    assert get_objective("Recall") in automl.additional_objectives
+
+
+@pytest.mark.parametrize("imbalanced_data", [True, False])
+def test_use_recommendation_score_imbalanced(
+    imbalanced_data,
+    mock_imbalanced_data_X_y,
+    X_y_multi,
+):
+    if imbalanced_data:
+        X, y = mock_imbalanced_data_X_y("multiclass", "none", "small")
+        X = X[
+            :-300
+        ]  # Make the dataset more imbalanced, so that the data check flags it
+        y = y[:-300]
+    else:
+        X, y = X_y_multi
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="multiclass",
+        use_recommendation=True,
+    )
+    objectives = get_default_recommendation_objectives(
+        "multiclass",
+        imbalanced=imbalanced_data,
+    )
+    assert automl.recommendation_objectives == objectives
+
+
+def test_use_recommendation_score(AutoMLTestEnv, X_y_binary):
```

Comment:
Hahaha, I can't say I'm in love with the architecture of this test....it's basically 4 tests in one, so if it fails, you still need to go and determine which of the internal tests is failing and why.  Additionally, it covers up any successive failures i.e. internal test #2 fails the test but internal test #4 is also failing.  I would challenging you to think of a way to maybe parametrize this so that the individually parameterized tests can fail and that the parameter reveals hints at the test condition that causes the failure.